marlowe/specification/ReadMe.md

```diff
@@ -0,0 +1,346 @@
+# On-Chain Transaction Specification for Marlowe
+
+
+## Scope
+
+This document defines the specification for Marlowe semantics's interface with the Cardano blockchain. Marlowe utilizes three Plutus scripts: (i) a monetary policy for the roles currency used in some Marlowe contracts; (ii) the Marlowe application validator script that enforces Marlowe semantics; and (iii) the Marlowe payout-validator script that allows the holder of a role token to withdraw funds paid by the Marlowe application.
+
+
+## Participants in Marlowe Contracts
+
+Marlowe contracts identify each participant by either a *public-key hash (PKH)* or a *role*. Marlowe semantics allow PKHs and roles to be freely mixed in contract logic.
+- A Marlowe PKH corresponds to the hash of a Cardano payment verification key, `Hash PaymentKey`.
+- A Marlowe role corresponds to a Cardano native token with the `TokenName`  for the role. (In Cardano, role names may be zero to thirty-two bytes long.) All of the role tokens in a given Marlowe contract must use the same  `CurrencySymbol` .
+
+
+## Life Cycle of a Marlowe Contract
+
+The execution of a Marlowe contract proceeds as a linear sequence of applications of inputs at the contract's script address.
+1. Role tokens are typically minted prior to or within the creation transaction of the contract, though this is not enforced on-chain.
+2. The creation transaction for the contract stores the state of the contract in the datum at the contract's script address.
+3. Each application of input to the contract updates the state/datum at that same address.
+4. When a contract closes, there is no output to the contract's script address.
+5. If the contract pays funds to a role during application of inputs, those funds are sent to the address of the Marlowe payout-validator script.
+
+Thus each Marlowe contract is a finite linear sequence of continuations at the script address, from creation to closure.
+
+
+## Monetary Policy for Role Tokens
+
+Any Cardano monetary policy may be used to mint the role tokens used in a Marlowe contract. For security in standard Marlowe use cases, a one-time or locked minting policy such as `Plutus.Contracts.Currency.OneShotCurrency` is recommended. Exotic use cases might employ other monetary policies. It is the responsibility both of the developer of the off-chain code managing a contract and also of the user of the contract to verify that the monetary policy of the role tokens meets their security requirements.
+
+
+## Representation of Marlowe Semantics in Plutus
+
+Marlowe semantics are embodied in the Haskell function `computeTransaction` that is generated from [the Isabelle implementation of the Marlowe DSL](https://github.com/input-output-hk/marlowe/tree/master/isabelle/Core). Note that `computeTransaction`  uses the `PlutusTx` types and functions instead of the standard-prelude or GHC-base Haskell ones.
```

Comment:
I was wondering about the "generated from" too.

Reply:
I'll also check with @hrajchert about this, since he is looking at Isabelle->Haskell code generation. There is a possibility that we could move from "translated" to "generated" soon.

Reply:
It is not generated, it just follows the Isabelle implementation. And after we implement the test oracle we can link to it to say how we make sure it follows it closely.

Reply:
Thanks for the clarifications.